### DISCUSSION TITLE
Add adapter classes for Microsoft.Extensions.AI

### DISCUSSION THREAD

**Discussion by Nukepayload2** [SciSharp/LLamaSharp#1022] [Ideas]
As an app developer, I need to let end users control where the LLM model should be running. For Llama models running on local computer in the application's process, we use LlamaSharp. If the user want to use an AI service on the Intranet, we use Ollama. If LlamaSharp implements https://www.nuget.org/packages/Microsoft.Extensions.AI.Abstractions , the model provider logics would be much easier to implement.

**Comment by martindevans** [comment]
Stephen Toub added support for these classes in https://github.com/SciSharp/LLamaSharp/pull/964, does that work for you or are there some bits missing?

**Reply by Nukepayload2** [reply]
Sorry, I mistakenly thought that the `Microsoft.Extensions.AI` integration doesn't exist, because I didn't find anything like `ILLamaExecutor.AsChatClient()` or `Microsoft.Extensions.AI` in the documentation.
